chromadb/test/property/strategies.py

```diff
@@ -39,46 +39,130 @@ def collections(draw) -> Collection:
     """Strategy to generate a set of collections"""
 
     # name = draw(st.from_regex(coll_name_re))
-    name = draw(st.one_of(st.from_regex(_coll_name_re)))
+    name = draw(st.one_of(st.from_regex(_collection_name_re)))
     hypothesis.assume(not _ipv4_address_re.match(name))
     hypothesis.assume(not _two_periods_re.search(name))
 
-    return {"name": name, "metadata": draw(metadata)}
+    return {"name": name, "metadata": draw(collection_metadata)}
+
+
+def one_or_both(strategy_a, strategy_b):
+    return st.one_of(
+        st.tuples(strategy_a, strategy_b),
+        st.tuples(strategy_a, st.none()),
+        st.tuples(st.none(), strategy_b),
+    )
+
+
+@st.composite
+def unique_ids_strategy(draw, count: int):
+
+    ratio = 20
+    strs = count // ratio
+
+    str_results = draw(
+        st.lists(st.text(min_size=1, max_size=64), min_size=strs, max_size=strs, unique=True)
+    )
+
+    # Rotate selections from between the two lists
+    results = []
+    for i in range(count):
+        if i % ratio == 0 and len(str_results) > 0:
+            results.append(str_results.pop())
+        else:
+            results.append(str(draw(st.uuids())))
+
+    return results
+
+
+float_types = [np.float16, np.float32, np.float64]
```

Comment:
Should we unify this with https://github.com/chroma-core/chroma/pull/324/files#diff-6374b894d196669ac1ce5b4cc486d6d5446a972cb09c78d10342b24343ab5549R77 in #324

Reply:
This was actually broken out from that because we have to generate floats and ints differently when using a strategy. Would prefer to keep it this way for now.